src/commands/Minion/favalch.ts

```diff
@@ -1,52 +1,79 @@
+import { MessageAttachment } from 'discord.js';
+import { chunk } from 'e';
 import { ArrayActions, CommandStore, KlasaMessage } from 'klasa';
-import { Item } from 'oldschooljs/dist/meta/types';
 
 import { UserSettings } from '../../lib/settings/types/UserSettings';
 import { BotCommand } from '../../lib/structures/BotCommand';
+import { ItemBank } from '../../lib/types';
 import { itemNameFromID } from '../../lib/util';
+import { parseStringBank } from '../../lib/util/parseStringBank';
 
 export default class extends BotCommand {
 	public constructor(store: CommandStore, file: string[], directory: string) {
 		super(store, file, directory, {
-			usage: '[item:item]',
+			usage: '[item:...string]',
 			description: 'Favorites an item to be used in alching.',
-			examples: ['+favalch rune platebody'],
+			examples: ['+favalch rune platebody', '+favalch rune platebody, dragon halberd, dragon longsword'],
 			categoryFlags: ['minion', 'skilling']
 		});
 	}
 
-	async run(msg: KlasaMessage, [items]: [Item[] | undefined]) {
+	async run(msg: KlasaMessage, [str]: [string]) {
 		const currentFavorites = msg.author.settings.get(UserSettings.FavoriteAlchables);
-
-		if (!items || items.length === 0) {
-			const currentFavorites = msg.author.settings.get(UserSettings.FavoriteAlchables);
-			if (currentFavorites.length === 0) {
-				return msg.channel.send('You have no favorited alchable items.');
+		const newFavorites: number[] = [...currentFavorites];
+		const items = parseStringBank(str);
+		if (!str || items.length === 0) {
+			if (currentFavorites.length === 0) return msg.channel.send('You have no favorited alchable items.');
+			if (msg.flagArgs.text) {
+				return msg.channel.send(
+					`Your current favorite alchable items are: ${chunk(
+						currentFavorites.map(id => itemNameFromID(id)),
+						50
+					)
+						.map(m => m.join(', '))
+						.join('\n')}.`
+				);
 			}
-			return msg.channel.send(
-				`Your current favorite alchable items are: ${currentFavorites
-					.map(id => itemNameFromID(id))
-					.join(', ')}.`
-			);
+			const imageBank: ItemBank = {};
+			for (const i of newFavorites) imageBank[i] = 0;
+			const { image } = await this.client.tasks
+				.get('bankImage')!
+				.generateBankImage(imageBank, 'Your current favorite alchable items', false, { id: 'yes' }, msg.author);
+			return msg.channel.send({ files: [new MessageAttachment(image!, 'youFavoriteItems.png')] });
 		}
 
-		const [item] = items;
-
-		if (!item.highalch) {
-			return msg.channel.send("That item isn't alchable.");
+		const dupeCheck: string[] = [];
+		const cantAlchItems: string[] = [];
+		const removed: string[] = [];
+		const added: string[] = [];
+		for (const [item] of items) {
```

Comment:
Again you can refactor this with a map(item => { 
//do stuff; 
return addItem ? objToAdd : undefined;
}).filter(item => item !== undefined);